Expose `WKPreferences/tabFocusesLinks` internally on iOS

rr-codes opened the pull request:
#### 22f1a23dde6d414c7c6e860d970cc3f8e038178a
<pre>
Expose `WKPreferences/tabFocusesLinks` internally on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=283651">https://bugs.webkit.org/show_bug.cgi?id=283651</a>
<a href="https://rdar.apple.com/140508437">rdar://140508437</a>

Reviewed by Wenson Hsieh.

The logic for having the tab key switch focus to forms and links is already platform-agnostic, so this
just exposes it for iOS.

Also de-duplicate some code.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::tabsToAllFormControls const):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::tabsToAllFormControls const): Deleted.
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::tabsToAllFormControls const): Deleted.
* Source/WebKit/Modules/Internal/WebKitInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.mm:
(-[WKPreferences encodeWithCoder:]):
(-[WKPreferences initWithCoder:]):
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesInternal.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Coding.mm:
(TEST(Coding, WKPreferences)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Copying.mm:
(TEST(Copying, WKPreferences)):

Canonical link: <a href="https://commits.webkit.org/287221@main">https://commits.webkit.org/287221@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cae73c22b2055c15b459409877bc19ef1ba971c5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78839 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/57883 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32221 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83497 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30100 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/80972 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67033 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6164 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/61744 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19674 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/81906 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51779 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/71694 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/42051 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49125 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/25924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28441 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70238 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/26333 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84868 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6204 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/4289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/69971 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6366 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/67771 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/69225 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17227 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/13268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/11995 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6149 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/12034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6132 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9569 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/7923 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/UIProcess/API/Cocoa/WKPreferencesInternal.h:
```diff
@@ -42,3 +44,13 @@ template<> struct WrapperTraits<WebPreferences> {
+
+@interface WKPreferences ()
+
+#if TARGET_OS_IPHONE
```
Nit - we prefer using `PLATFORM()` when possible.

```suggestion
#if PLATFORM(IOS_FAMILY)
```

rr-codes replied:
ah, I forgot we can do that in Internal headers!

rr-codes committed:
Expose `WKPreferences/tabFocusesLinks` internally on iOS
https://bugs.webkit.org/show_bug.cgi?id=283651
rdar://140508437

Reviewed by Wenson Hsieh.

The logic for having the tab key switch focus to forms and links is already platform-agnostic, so this
just exposes it for iOS.

Also de-duplicate some code.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::tabsToAllFormControls const):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::tabsToAllFormControls const): Deleted.
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::tabsToAllFormControls const): Deleted.
* Source/WebKit/Modules/Internal/WebKitInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.mm:
(-[WKPreferences encodeWithCoder:]):
(-[WKPreferences initWithCoder:]):
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesInternal.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Coding.mm:
(TEST(Coding, WKPreferences)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Copying.mm:
(TEST(Copying, WKPreferences)):

Canonical link: https://commits.webkit.org/287221@main
